app/code/Magento/ImportExport/Controller/Adminhtml/Export/File/Download.php

```diff
@@ -55,12 +54,17 @@ public function __construct(
      * Controller basic method implementation.
      *
      * @return \Magento\Framework\App\ResponseInterface
-     * @throws LocalizedException
      */
     public function execute()
     {
-        if (empty($fileName = $this->getRequest()->getParam('filename'))) {
-            throw new LocalizedException(__('Please provide export file name'));
+        /** @var \Magento\Framework\Controller\Result\Redirect $resultRedirect */
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $resultRedirect->setPath('adminhtml/export/index');
+        $fileName = $this->getRequest()->getParam('filename');
+        if (empty($fileName) || \preg_match('/\.\.(\\\|\/)/', $fileName) !== 0) {
```

Comment:
Could you provide more on what exactly does this pregmatch check?

```suggestion
        if (empty($fileName) || preg_match('/\.\.(\\\|\/)/', $fileName) !== 0) {
```

Reply:
Hi @eduard13  I didn't modified this regex but it looks like the original author checks if $fileName is not a ".."

Reply:
Hi @pmarki, I think that this check should be handled by `$directory->isFile($path)`.  I suggest removing it and keep it simple.
Also, could you please cover these fixes by a Unit Test?

Thank you.

Reply:
Hi @eduard13 I investigated it a bit more and this preg_match prevents path traversal. Without it admin user with export permissions would be able to download any file located in var folder. 
I checked the DeleteController and I was able to delete any file in var folder as well so I added this check to Delete Controller. 

Unit tests added.

Reply:
Awesome, thank you.